connect.go

```diff
@@ -358,15 +358,18 @@ type handlerConnCloser interface {
 // envelopes the message and attaches headers and trailers. It attempts to
 // consume the response stream and isn't appropriate when receiving multiple
 // messages.
-func receiveUnaryResponse[T any](conn StreamingClientConn) (*Response[T], error) {
+func receiveUnaryResponse[T any](conn StreamingClientConn, config *clientConfig) (*Response[T], error) {
 	var msg T
+	if err := config.Initializer(conn.Spec(), &msg); err != nil {
+		return nil, err
+	}
 	if err := conn.Receive(&msg); err != nil {
 		return nil, err
 	}
 	// In a well-formed stream, the response message may be followed by a block
 	// of in-stream trailers or HTTP trailers. To ensure that we receive the
 	// trailers, try to read another message from the stream.
-	if err := conn.Receive(new(T)); err == nil {
+	if err := conn.Receive(nil); err == nil {
```

Comment:
Made it nil to avoid calling initializer twice, should never be used to unmarshal into.

Reply:
Could also pass the message again?

Reply:
If this succeeds (like if there _is_ another message on the stream), this seems like a source of likely nil-dereference panics. Please don't change this to nil.

I think it's safest as it was and just call the initializer 2x. Maybe just add a TODO about optimizing away the allocation and second initialization (like an alternate method on the streaming conn that could be used to test if there is more data on the stream, or push this check for a second message into the streaming conn, so it can done without allocating and providing a destination message).